smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/json/MemberShapeEncodeGenerator.kt

```diff
@@ -164,8 +164,8 @@ abstract class MemberShapeEncodeGenerator(
         when (targetShape) {
             is CollectionShape -> {
                 val topLevelContainerName = "${memberName}Container"
-                writer.write("var \$L = $containerName.nestedContainer(keyedBy: \$N.self)", topLevelContainerName, ClientRuntimeTypes.Serde.Key)
-                renderEncodeMap(ctx, memberName, topLevelContainerName, targetShape, level)
+                writer.write("var \$L = $containerName.nestedUnkeyedContainer(forKey: \$N($dictKey${level - 1}))", topLevelContainerName, ClientRuntimeTypes.Serde.Key)
```

Comment:
is it possible for level to be zero?  if so, then we would generate `Key(dictKey-1)` ?

Reply:
not for it to go here in the when statement. level would have to be >= 1 if the target shape were a collection shape

Reply:
but I did think of that and check that that wasn't gonna happen :)

Reply:
I think it would make the code more readable if we removed the default value of ` = 0` for this parameter -- and then in StructEncodeGenerator, pass in "0" so that i the call to this method looks like:
```
                                renderEncodeMapMember(target, memberName, containerName, 0)
```